scripts/rescript_format.js

```diff
@@ -55,12 +67,56 @@ async function readStdin() {
   return Buffer.concat(chunks).toString("utf8");
 }
 
+/**
+ * @param {string[]} files
+ * @param {string} bsc_exe
+ * @param {(x: string) => boolean} isSupportedFile
+ * @param {boolean} checkFormatting
+ */
+async function formatFiles(files, bsc_exe, isSupportedFile, checkFormatting) {
+  var incorrectlyFormattedFiles = 0;
+  try {
+    const _promises = await Promise.all(
+      files.map(async file => {
+        if (isSupportedFile(file)) {
+          // console.log(`processing ${arg}`);
+          const flags = checkFormatting
+            ? ["-format", file]
+            : ["-o", file, "-format", file];
+          const { stdout } = await asyncExecFile(bsc_exe, flags);
+          if (check.val) {
+            const original = await asyncFs.readFile(file, "utf-8");
+            if (original != stdout) {
+              console.error("[format check]", file);
+              incorrectlyFormattedFiles++;
+            }
+          }
+        }
+        return null;
+      })
+    );
+  } catch (err) {
+    console.error(err);
+    process.exit(2);
+  }
+  if (incorrectlyFormattedFiles > 0) {
+    console.error(
+      `${incorrectlyFormattedFiles} file${
+        incorrectlyFormattedFiles > 1 ? "s" : ""
```

Comment:
Maybe more readable to just do a separate console.error for 1 file vs. n files.

Reply:
indeed!